src/main/java/org/openrewrite/java/migrate/io/RemoveFinalizeFromFileStream.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate.io;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveFinalizeFromFileStream extends Recipe {
+
+    private static final String JAVA_IO_FILEINPUTSTREAM = "java.io.FileInputStream";
+    private static final String JAVA_IO_FILEOUTPUTSTREAM = "java.io.FileOutputStream";
+
+    private static final MethodMatcher METHOD_MATCHER = new MethodMatcher("java.lang.Object finalize()");
```

Comment:
Instead I suggest we add two specific method matchers like `new MethodMatcher("java.io.FileInputStream finalize()", true)` and `new MethodMatcher("java.io. FileOutputStream finalize()", true)`.

Reply:
@knutwannheden  

As mentioned, I've made the needed changes to the recipe but `matches` isn't working for me as expected for deprecated methods which is why I had taken this approach. Let me know if you have any inputs for making this work. I committed the code using MethodMatcher as suggested causing test failure for me. I tried similar approach for other pull requests(#244 , #247 ) that I've created but haven't been successful, so took this approach. 

Getting the below error in the test execution:
```
> Task :test
line 1:25 mismatched input ' ' expecting {'(', '*'}
FooBar.java:3: error: no suitable constructor found for FileInputStream(no arguments)
class FooBar extends FileInputStream{
^
    constructor FileInputStream.FileInputStream(String) is not applicable
      (actual and formal argument lists differ in length)
    constructor FileInputStream.FileInputStream(File) is not applicable
      (actual and formal argument lists differ in length)
    constructor FileInputStream.FileInputStream(FileDescriptor) is not applicable
      (actual and formal argument lists differ in length)

Expected recipe to complete in 1 cycle, but took 0 cycle.
java.lang.AssertionError: Expected recipe to complete in 1 cycle, but took 0 cycle.
	at org.openrewrite.test.LargeSourceSetCheckingExpectedCycles.afterCycle(LargeSourceSetCheckingExpectedCycles.java:80)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:88)
	at org.openrewrite.Recipe.run(Recipe.java:279)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:326)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:130)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:125)
	at org.openrewrite.java.migrate.io.RemoveFinalizeFromFileStreamTest.removeFinalizerForFileInputStream(RemoveFinalizeFromFileStreamTest.java:36)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

```

Reply:
Tried this myself as well with two separate matchers, but that failed; Only the matcher for `Object finalize()` correctly matches, so we keep the TypeUtils check.